lib/mayaUsd/utils/utilFileSystem.cpp

```diff
@@ -117,6 +117,30 @@ std::string UsdMayaUtilFileSystem::getMayaSceneFileDir()
     return std::string();
 }
 
+std::string UsdMayaUtilFileSystem::getPathRelativeToMayaSceneFile(const std::string& fileName)
+{
+    ghc::filesystem::path absolutePath(fileName);
+    ghc::filesystem::path basePath(getMayaSceneFileDir());
+    ghc::filesystem::path relativePath = absolutePath.lexically_relative(basePath);
+
+    if (relativePath.empty()) {
+        TF_RUNTIME_ERROR(
+            "File name (%s) cannot be resolved as relative to the Maya scene file, using the "
+            "absolute path.",
+            fileName.c_str());
```

Comment:
runtime_error seems pretty strong in this case. You've detected the problem and are returning a good value. An error would make it appear as if the user did something wrong, but there is nothing they can do about it here. So I think a warning is sufficient.